Source/WebCore/html/HTMLLabelElement.cpp

```diff
@@ -140,6 +143,15 @@ bool HTMLLabelElement::isEventTargetedAtInteractiveDescendants(Event& event) con
 void HTMLLabelElement::defaultEventHandler(Event& event)
 {
     if (event.type() == eventNames().clickEvent && !m_processingClick) {
+        // If text of label element is selected, do not pass the event to control element.
+        // Note: a click event may be not a mouse event if created by document.createEvent().
+        if (event.isMouseEvent() && !downcast<MouseEvent>(event).isSimulated()) {
+            if (auto* frame = document().frame()) {
+                if (frame->selection().selection().isRange())
+                return;
```

Comment:
Might be more readable as:

```
auto* frame = document().frame();
if (frame && frame->selection().selection().isRange())
    return;
```

To reduce the nesting.

Reply:
Done.